### PR TITLE
Shiny type

### PR DESCRIPTION
**Pull request by yan130**
## Description
Update MET products on the shiny app ( type input)
Add models to BrownDog shiny app 
Update shiny web app to use BrownDog R library. 
Update met.process.R for convert .xml to .pecan.nc/ .pecan.zip

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: shiny/BrownDog/server.R [hunk 1]
```diff
@@ -41,87 +85,130 @@ server <- shinyServer(function(input, output, session) {
...
       selectedsite <- reactive({
-        if(!is.null(input$agreement) && input$agreement){
-          paste(c("<input>", paste0("  <type>", input$type, "</type>"), 
-                  paste0("  <site>", click$id, "</site>"),
-                  paste0("  <lat>", click$lat, "</lat>"),
-                  paste0("  <lon>", click$lng, "</lon>"),
-                  paste0("  <start_date>", if (input$start_date != "") input$start_date else "2001", "-01-01 00:00:00</start_date>"),
-                  paste0("  <end_date>", if (input$end_date != "") input$end_date else "2001", "-12-31 23:59:59</end_date>"),
-                  "</input>"), collapse="\n")
+        if (!is.element(input$type, c("AmeriFlux", "NARR", "FLUXNET")) ||
```
Comment (robkooper) [inline shiny/BrownDog/server.R]: Use +          if (is.null(input$agreement) || input$agreement) {

File: shiny/BrownDog/server.R [hunk 2]
```diff
@@ -41,87 +85,130 @@ server <- shinyServer(function(input, output, session) {
...
-          url <- gsub('.*<a.*>(.*)</a>.*', '\\1', result_bds)
-          wait <- 300
-          #delet xml file
-          unlink(xml_filename)
-          browndog.download(url[1], file, token, wait)
+          library(BrownDog)
+          bds <- "https://bd-api-dev.ncsa.illinois.edu"
+          output_path    <- "/tmp/"
+          url <-
+            convert_file(
```
Comment (robkooper) [inline shiny/BrownDog/server.R]: Use BrownDog::convert_file

File: shiny/BrownDog/server.R [hunk 3]
```diff
@@ -41,87 +85,130 @@ server <- shinyServer(function(input, output, session) {
...
-          httpheader <- c(Accept="text/plain", Authorization = input$token)
-          curloptions <- list(httpheader = httpheader)
-          result_bds <- postForm(convert_api,"file"= fileUpload(xml_filename),.opts = curloptions)
-          print(result_bds)
-          url <- gsub('.*<a.*>(.*)</a>.*', '\\1', result_bds)
-          wait <- 300
-          #delet xml file
-          unlink(xml_filename)
-          browndog.download(url[1], file, token, wait)
+          library(BrownDog)
```
Comment (robkooper) [inline shiny/BrownDog/server.R]: remove BrownDog

**Comment by robkooper** [comment]
Make sure BrownDog is in the Description.

@ashiklom what is the best way to list dependencies for shiny, should we add a special target?

**Comment by ashiklom** [comment]
I don't know if Shiny apps obey typical R package installation rules, but my guess is no. I see a few options:

1. Create a dummy Shiny package containing Shiny dependencies, and add that to the makefile. 
2. Add the Shiny dependencies to the visualization package. 
3. Add explicit Shiny dependencies into the makefile, like we do with devtools. 

My preference would be for 1 or 2 because devtools::install is, I believe, smart enough to update packages if they're out of date. 2 requires the fewest changes, but is arguably the least intuitive.

**Comment by robkooper** [comment]
I think I would vote for option 1. We already have a shiny folder and we can add the DESCRIPTION file there with the dependencies.

The other option is to make this the first separate repo and have a single repo with all the shiny modules.

**Comment by tonygardella** [comment]
@yan130 Could you compare your pull request to the "Develop" branch rather than the "Master" branch?

**Comment by yan130** [comment]
@robkooper I am not authorized to merge this pull request.

**Pull request merged**